q2_alignment/plugin_setup.py

```diff
@@ -40,6 +42,37 @@
     citations=[citations['katoh2013mafft']]
 )
 
+plugin.methods.register_function(
+    function=q2_alignment.sina,
+    inputs={'sequences': FeatureData[Sequence],
+            'reference': FeatureData[AlignedSequence]},
+    parameters={'arb_reference': Str,
+                'num_references': Int % Range(0, None),
+                'kmer_len': Int % Range(5, 25)},
+    outputs=[('alignment', FeatureData[AlignedSequence])],
+    input_descriptions={
+        'sequences': 'The sequences to be aligned.',
+        'reference': 'The reference alignment (QZA format).',
+    },
+    parameter_descriptions={
+        'arb_reference': 'The reference alignment (ARB format).',
```

Comment:
There was some discussion of this on the forum, but just want to bring it up in the context of the review so we have it here. 

Providing a file path as a parameter is generally not something you want to do in QIIME 2 for a couple of reasons. First, there won't be very useful provenance for this input possibly resulting in ambiguity over exactly what reference was used in analysis. Next, with some interfaces (e.g., a GUI) providing the arb database as a parameter will be pretty inconvenient for users - because it's a parameter, they wouldn't get a file select dialog but rather would have to type the absolute path to this file (or worse, with a web interface, it may not even be possible). 

Ultimately, we should create a new `DirectoryFormat` and semantic type for ARB reference database, so we can have useful provenance information about these, and accept them as inputs rather than parameters. This would be ideal to do before merge, but if that's not possible, would you mind creating an issue on the q2-alignment issue tracker to do this?

Reply:
I went for the "simplest hack" with the parameter. Adding a semantic type / format for ARB databases didn't seem like a good solution at this time either, as it would require time consuming unpacking and indexing stages for typically large reference databases on every invocation.

Mid-term (~months), SINA will drop the ARB file requirement for indices, so perhaps we can get around needing the ARB type at all. The catch is that I'll have to create a new file format for storing very large alignments. An alignment comprising 1M sequences and 50k columns in FASTA format is simply too large, and takes an inordinate amount of time to parse. I don't like adding yet another format to our world, but I don't see an alternative (would be happy to hear your thoughts if you're interested in this).

So in summary - perhaps create an issue and hopefully close it if/when ARB as a type can be avoided.

Reply:
Thanks @epruesse. Would you mind creating this issue on q2-alignment?